### ISSUE TITLE
needless_borrow suggesting the same fix as exists in code

### ISSUE THREAD

**Issue by dessalines** [rust-lang/rust-clippy#7590]
Lint name: needless_borrow

I tried this code, which is getting a connection from a deadpool database pool:

```rust
let join = CommunityModerator::join(&&context.pool.get().await?, &community_moderator_form);
```

I expected to see this happen: No clippy warnings.

Instead, this happened: 
```
error: this expression borrows a reference (`&deadpool_diesel::connection::Connection<diesel::pg::connection::PgConnection>`) that is immed
iately dereferenced by the compiler
   --> crates/api_crud/src/user/create.rs:195:43
    |
195 |       let join = CommunityModerator::join(&&context.pool.get().await?, &community_moderator_form);
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: change this to: `&&context.pool.get().await?`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
```
Its suggesting a fix that's the same as the error

### Meta

- `cargo clippy -V`: clippy 0.1.56 (ae90dcf 2021-08-09)
- `rustc -Vv`:
  
```
rustc 1.56.0-nightly (ae90dcf02 2021-08-09)
binary: rustc
commit-hash: ae90dcf0207c57c3034f00b07048d63f8b2363c8
commit-date: 2021-08-09
host: x86_64-unknown-linux-gnu
release: 1.56.0-nightly
LLVM version: 12.0.1
```

**Comment by DevinR528** [comment]
Yesh, after downloading/installing libpq I was able to confirm that this no longer is a problem.

```
warning: this expression borrows a reference (`&diesel::r2d2::PooledConnection<diesel::r2d2::ConnectionManager<diesel::PgConnection>>`) that is immediately dereferenced by the compiler
   --> crates/apub/src/fetcher/community.rs:108:43
    |
108 |       let join = CommunityModerator::join(&&context.pool.get()?, &community_moderator_form);
    |                                           ^^^^^^^^^^^^^^^^^^^^^ help: change this to: `&context.pool.get()?`
    |
    = note: `#[warn(clippy::needless_borrow)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow

warning: `lemmy_apub` (lib) generated 2 warnings
```
The only difference is it's not async but I'm not sure if that's a problem, the way the lints set up I doubt it?

**Comment by dessalines** [comment]
One piece of context I can give, is that in order to get a usable `&PgConnection` from deadpools `Connection<PgConnection>`, you have to dereference it twice:

```rust
    let test : Connection<PgConnection> = context.pool.get().await?;
    let test2: &PgConnection = &&test;
```

**Comment by DevinR528** [comment]
Hmm, yeah, although it should either fail the same way for `&&Arc<Arc<Thing>>`  as an argument to an fn that takes `&Thing` right?
